modules/extensions/src/views/lite/components/devtools/index.tsx

```diff
@@ -0,0 +1,113 @@
+import { Button, ButtonGroup, Tab, Tabs } from '@blueprintjs/core'
+import '@blueprintjs/core/lib/css/blueprint.css'
+import nanoid from 'nanoid'
+import React from 'react'
+import { MdBugReport } from 'react-icons/md'
+
+import Settings from './settings'
+import style from './style.scss'
+import { Decision } from './views/Decision'
+import { Entities } from './views/Entities'
+import { Inspector } from './views/Inspector'
+import { Intents } from './views/Intents'
+import { Slots } from './views/Slots'
+import { Suggestions } from './views/Suggestions'
+import Header from './Header'
+import SplashScreen from './SplashScreen'
+export const updater = { isLoaded: false, callback: undefined }
+
+const WEBCHAT_WIDTH = 350
+const DEV_TOOOLS_WIDTH = 450
+
+export class DevTools extends React.Component<DevToolProps, DevToolState> {
+  state = {
+    event: undefined,
+    visible: false,
+    selectedTabId: 'basic',
+    showSettings: false
+  }
+
+  componentDidMount() {
+    updater.callback = data => {
+      this.setState({ event: data })
+    }
+
+    this.props.store.view.setLayoutWidth(WEBCHAT_WIDTH)
+    this.props.store.view.setContainerWidth(WEBCHAT_WIDTH + DEV_TOOOLS_WIDTH)
+    this.props.store.setMessageWrapper({ module: 'extensions', component: 'Wrapper' })
+
+    this.props.store.view.addHeaderButton({
```

Comment:
couldn't we use a portal (like the keyboard composer) instead of having "add/remove XXX" ?  this way you would simply in the render() method do:

```ts
<WebchatHeader.Button label="Show Debugger" icon={} onClick={}>
  <.........>
</WebchatHeader.Button>
```

You wouldn't have the need for the "update XXX" method I see commented below too

Reply:
That could be an idea, however we would need to create a "Webchat SDK" that can be imported by modules for typings. 

It would require a bit more work, since it would need to check portals in custom messages, and all overridable component. Right now, the store is made available to all of them, so it's easier to use (i think)